Javadoc generation fails with JDK 8

```
11 warnings
[INFO] ------------------------------------------------------------------------
[ERROR] BUILD ERROR
[INFO] ------------------------------------------------------------------------
[INFO] MavenReportException: Error while creating archive:
Exit code: 1 - /Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/config/Option.java:61: warning: no @param for <T>
    public <T> boolean hasSameNameAs(Option<T> option) {
                       ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/config/Option.java:21: warning: no description for @param
 * @param <T>
   ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/material/packagerepository/PackageRevision.java:120: warning: no @param for key
    public String getDataFor(String key) {
                  ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/Console.java:35: error: self-closing element not allowed
     * <p/>
       ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/Console.java:38: error: self-closing element not allowed
     * <p/>
       ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/Console.java:47: error: self-closing element not allowed
     * <p/>
       ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/Console.java:50: error: self-closing element not allowed
     * <p/>
       ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/EnvironmentVariables.java:33: warning: no @param for console
    void writeTo(Console console);
         ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskConfig.java:45: warning: no @param for propertyName
    public String getValue(String propertyName) {
                  ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskConfig.java:45: warning: no @return
    public String getValue(String propertyName) {
                  ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskExecutionContext.java:27: warning: no @return
    EnvironmentVariables environment();
                         ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskExecutionContext.java:33: warning: no @return
    Console console();
            ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskExecutionContext.java:38: warning: no @return
    String workingDir();
           ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskView.java:27: warning: no @return
    String displayValue();
           ^
/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskView.java:32: warning: no @return
    String template();
           ^

Command line was: /Library/Java/JavaVirtualMachines/jdk1.8.0.jdk/Contents/Home/bin/javadoc @options @packages

Refer to the generated Javadoc files in '/Users/rjunior/code/github/gocd/plugin-infra/go-plugin-api/target/javadoc' dir.


[INFO] ------------------------------------------------------------------------
[INFO] For more information, run Maven with the -e switch
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2 minutes 46 seconds
[INFO] Finished at: Fri Apr 25 19:18:48 BRT 2014
[INFO] Final Memory: 61M/465M
[INFO] ------------------------------------------------------------------------
Buildr aborted!
Failed to run: mvn install -DskipTests

(See full trace by running task with --trace)
```

--- a/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/config/Option.java
+++ b/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/config/Option.java
@@ -18,7 +18,7 @@ package com.thoughtworks.go.plugin.api.config;
 
 /**
  * Option could be used to specify metadata for a Property.
- * @param <T>
+ * @param <T> type for the option value
  */
 public class Option<T> {
     private String name;
@@ -55,6 +55,7 @@ public class Option<T> {
 
     /**
      * Checks if this option has same name as the provided option
+     * @param <T> type for the option value
      * @param option the option for which name equality has to be checked
      * @return true if name matches
      */

--- a/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/material/packagerepository/PackageRevision.java
+++ b/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/material/packagerepository/PackageRevision.java
@@ -115,6 +115,7 @@ public class PackageRevision {
 
     /**
      * Gets additional data related to package revision for given key
+     * @param key for additional data
      * @return additional data related to package revision for given key
      */
     public String getDataFor(String key) {

--- a/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/Console.java
+++ b/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/Console.java
@@ -32,10 +32,10 @@ public interface Console {
 
     /**
      * Setup the console to read the input stream as standard error.
-     * <p/>
+     * <p></p>
      * This is used to connect the output of a process, to the build log. This is usually used as:
      * console.readErrorOf(process.getErrorStream());
-     * <p/>
+     * <p></p>
      * where the "process" object is of type {@link java.lang.Process}.
      *
      * @param in The input stream to read as standard error.
@@ -44,10 +44,10 @@ public interface Console {
 
     /**
      * Setup the console to read the input stream as standard output.
-     * <p/>
+     * <p></p>
      * This is used to connect the output of a process, to the build log. This is usually used as:
      * console.readOutputOf(process.getInputStream());
-     * <p/>
+     * <p></p>
      * where the "process" object is of type {@link java.lang.Process}.
      *
      * @param in The input stream to read as standard output.

--- a/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/EnvironmentVariables.java
+++ b/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/EnvironmentVariables.java
@@ -29,6 +29,7 @@ public interface EnvironmentVariables {
 
     /**
      * Write the environment variables to a {@link com.thoughtworks.go.plugin.api.task.Console}.
+     * @param console to write
      */
     void writeTo(Console console);
 

--- a/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskConfig.java
+++ b/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskConfig.java
@@ -40,7 +40,10 @@ public class TaskConfig extends Configuration {
     }
 
     /**
-     * The value of the specified property, or null if not found.
+     * Retrieves the value of a property.
+     * @param propertyName Name of the property (or key) in the configuration.
+     *
+     * @return the value of the specified property, or null if not found.
      */
     public String getValue(String propertyName) {
         Property property = super.get(propertyName);

--- a/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskExecutionContext.java
+++ b/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskExecutionContext.java
@@ -23,17 +23,20 @@ package com.thoughtworks.go.plugin.api.task;
 public interface TaskExecutionContext {
     /**
      * The environment variables that this task should run with.
+     * @return the environment variables that this task should run with.
      */
     EnvironmentVariables environment();
 
     /**
      * This Console instance will be setup such that any messages written to it will show
      * up in the build log.
+     * @return the console instance that any messages will be written to
      */
     Console console();
 
     /**
      * Working directory to be used by the task.
+     * @return the working directory to be used by the task.
      */
     String workingDir();
 }

--- a/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskView.java
+++ b/plugin-infra/go-plugin-api/src/com/thoughtworks/go/plugin/api/task/TaskView.java
@@ -23,11 +23,15 @@ public interface TaskView {
     /**
      * Specifies the display value of this task plugin. This value is used in the job UI's task dropdown
      * as well as in the title of the task definition dialog box.
+     *
+     * @return display value for the task plugin
      */
     String displayValue();
 
     /**
      * The template for the task configuration, written using Angular.js templating language.
+     *
+     * @return Angular.js template for the task configuration
      */
     String template();
 }

